Make banking data gem mandatory

magicjascha opened the pull request:
I'm not completely sure if this is the right way to do this, so I tell you the story and you tell me:

I searched where we use the banking_data gem in our mothership and couldn't find the code, however if you remove it, [errors are thrown](https://github.com/gapfish/gapfish/pull/9422). It turned out it was used in one of our dependencies  - [a modified branch of the published gem iban-tools](https://github.com/boostify/iban-tools/tree/banking_data). This modified gem was not specifying and pulling the dependency banking_data itself. [Part of the functionality was relying on it](https://github.com/boostify/iban-tools/blob/1b284b84d8bbab3d1d7454cdbcbf28acadc87d17/lib/iban-tools/conversion.rb#L42), but if it was not installed it didn't throw error, it just returned nil. Only in our system it started throwing errors when it was missing. 

It took me quite some time and the help of Schasse to understand what was happening and that's why I thought it shouldn't be like that. So to clean up the situation I forked the gem, merged the modification branch, which contained the usage of the banking_data gem into master, made the banking_data gem mandatory and declared it as a legitimate dependency, i.e. we can remove it from the user and support gemfile now.

Now we have this fork, which I will then start using in my ruby3 branch. 

I feel this fork is better than a branch with a hidden dependency, but I'm not experienced in the forking business. 

The result looks a little messy to me (not sure, maybe its ok) but that is independent of the above story: Since I'm not using a [published version of the banking_data gem](https://rubygems.org/gems/banking_data) right now, but the[ gapfish-fork](https://github.com/gapfish/banking_data/pull/1), half of the dependencies is now declared in a gemspec and the other half is using a Gemfile (which overwrites the gemspec, if there's conflicts). If we know that we want to keep updating [the original open source repo](https://github.com/opahk/banking_data) and publish the banking_data gem, we could of course use the real gem, that was not clear to me.

magicjascha committed:
make banking_data gem mandatory

magicjascha committed:
fix appraisal for travis

magicjascha committed:
use banking_data gem from gapfish repository

schasse commented:
* it would be best if we could move the changes to the upstream repository, but this didn't happen for 9 years and I don't think it's feasible at the moment.
* +1 for changing the dependencies in the gemspec makes it more readable
* forking gives us the access rights we need and the MIT license is good to do so

magicjascha committed:
remove gemspec dependency

The gemspec was not overwritten by the Gemfile as assumed, when this repository was used by other repsitories. They where using the published gems, i.e. not the newest version.
